### PR TITLE
[ampproject/amphtml] ♿ [Story a11y] Accessible close buttons

### PR DESCRIPTION
**Pull request by processprocess**
Contributes to #32493
Applies same pattern used in share menu and Twitter overlay's close button.

- Adds role=button back to close button elements ([context](https://github.com/ampproject/amphtml/pull/32863#pullrequestreview-598814942)).
- Adds accessible close button for inline-attachment.
- Fixes typo.

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-page-attachment.js [hunk 1]
```diff
@@ -105,12 +107,21 @@ export class AmpStoryPageAttachment extends DraggableDrawer {
...
+    const closeButtonEl = this.headerEl_.appendChild(
       htmlFor(this.element)`
-          <span class="i-amphtml-story-page-attachment-close-button" aria-label="X"
+          <button class="i-amphtml-story-page-attachment-close-button" aria-label="close"
               role="button">
-          </span>`
+          </button>`
     );
+    const storyEl = closest(this.element, (el) => el.tagName === 'AMP-STORY');
+    const localizationService = getLocalizationService(devAssert(storyEl));
```
Comment (mszylkowski) [inline extensions/amp-story/1.0/amp-story-page-attachment.js:117]: You may be able to pass any element to `getLocalizationService` as it uses the ampDoc to reference the service, saving you the `closest()` call

**Comment by mszylkowski** [comment]
Having a button with `role="button"` is what @gmajoulet was referring to in one of the comments? That would only make sense if there's sth parsing for the role attribute specifically

File: extensions/amp-story/1.0/amp-story-draggable-drawer-header.css [hunk 2]
```diff
@@ -40,6 +40,10 @@
   background-repeat: no-repeat !important;
   color: rgba(0, 0, 0, 0.87) !important;
   cursor: pointer !important;
+  /* default button style overrides */
+  box-sizing: content-box!important;
+  border: none;
+  background-color: transparent!important;
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-draggable-drawer-header.css:46]: Nit: space before `!important`, and actual `!important` on the border rule

File: extensions/amp-story/1.0/amp-story-page-attachment.js (hunk 1 above)
Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story-page-attachment.js:117]: +1, `this.element` should work?

**Review by gmajoulet — APPROVED** [review]

**Comment by gmajoulet** [comment]
> Having a button with role="button" is what @gmajoulet was referring to in one of the comments? That would only make sense if there's sth parsing for the role attribute specifically

Yes there is, it's very easy to break and we will change it asap, it tripped us many times in the past few weeks (for reference: https://github.com/ampproject/amphtml/blob/76cb240bbd5c965449dba934fed9265f375c2693/extensions/amp-story/1.0/page-advancement.js#L439-L462)

**Review by mszylkowski — APPROVED** [review]

**Review by Enriqe — APPROVED** [review]

**Pull request merged**